wandb/sdk/launch/registry/elastic_container_registry.py

```diff
@@ -0,0 +1,126 @@
+        """Initialize the Elastic Container Registry.
+
+        Args:
+            repo_name (str): The name of the repository.
+            environment (AwsEnvironment): The AWS environment.
+
+        Raises:
+            LaunchError: If there is an error verifying the registry.
+        """
+        super().__init__()
+        self.repo_name = repo_name
+        self.environment = environment
+        self.verify()
+
+    def verify(self) -> None:
+        """Verify that the registry is accessible and the configured repo exists.
+
+        Raises:
+            RegistryError: If there is an error verifying the registry.
+        """
+        try:
+            session = self.environment.get_session()
+            client = session.client("ecr")
+            response = client.describe_registry()
+            response = client.describe_repositories(repositoryNames=[self.repo_name])
+            self.uri = response["repositories"][0]["repositoryUri"].split("/")[0]
+
+        except botocore.exceptions.ClientError as e:
+            code = e.response["Error"]["Code"]
+            msg = e.response["Error"]["Message"]
+            # TODO: Log the code and the message here?
+            raise LaunchError(
+                f"Error verifying Elastic Container Registry: {code} {msg}"
+            )
+
+    def get_username_password(self) -> Tuple[str, str]:
+        """Get the username and password for the registry.
+
+        Returns:
+            (str, str): The username and password.
+
+        Raises:
+            RegistryError: If there is an error getting the username and password.
+        """
+        try:
+            session = self.environment.get_session()
+            client = session.client("ecr")
+            response = client.get_authorization_token()
+            username, password = base64.standard_b64decode(
+                response["authorizationData"][0]["authorizationToken"]
+            ).split(b":")
+            return username.decode("utf-8"), password.decode("utf-8")
+
+        except botocore.exceptions.ClientError as e:
+            code = e.response["Error"]["Code"]
+            msg = e.response["Error"]["Message"]
+            # TODO: Log the code and the message here?
+            raise LaunchError(f"Error getting username and password: {code} {msg}")
+
+    def get_repo_uri(self) -> str:
+        """Get the uri of the repository.
+
+        Returns:
+            str: The uri of the repository.
+        """
+        return self.uri + "/" + self.repo_name
+
+    def check_image_exists(self, image_uri: str) -> bool:
+        """Check if an image exists in the registry.
+
+        Args:
+            image_uri (str): The image uri.
+
+        Returns:
+            bool: True if the image exists, False otherwise.
+
+        Raises:
+            RegistryError: If there is an error checking if the image exists.
+        """
+        session = self.environment.get_session()
+        client = session.client("ecr")
```

Comment:
Same as before, we just need to make sure a region is passed. Instance roles on EC2 can create a valid session with creds but no region when calling `boto3.Session()`